src/components/Receiving/ItemDetails.js

```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
 
   render() {
-    const { close, itemList } = this.props;
+    const { close, itemList, locations } = this.props;
+    const poLineNumber = get(itemList, [0, 'poLineNumber'], '');
+    const title = get(itemList, [0, 'title'], '');
     const resultsFormatter = {
-      'title': line => get(line, 'title', ''),
-      'barcode': () => <TextField type="number" />,
-      'location': () => <Select fullWidth />,
-      'itemStatus': () => <Select fullWidth />,
+      'isChecked': (item) => (
+        <Checkbox
+          type="checkbox"
+          onChange={() => this.toggleItem(item)}
+          checked={!!(this.state.checkedItems.filter(el => el.id === item.id).length)}
+        />
+      ),
+      'barcode': () => (
+        <div className={css.editableFieldWrapper}>
+          <TextField type="number" />
+        </div>
+      ),
+      'format': (item) => <KeyValue value={get(item, 'orderFormat', '')} />,
+      'location': () => (
+        <div className={css.editableFieldWrapper}>
+          <Select
+            dataOptions={getLocationsForSelect(locations)}
+            defaultValue=""
+            fullWidth
+            placeholder=" "
+          />
+        </div>
+      ),
+      'itemStatus': () => (
+        <div className={css.editableFieldWrapper}>
+          <Select
+            placeholder=" "
+            defaultValue=""
+            fullWidth
+          >
+            {Object.keys(ITEM_STATUS).map((key) => (
+              <FormattedMessage
+                id={`ui-orders.receiving.itemStatus.${key}`}
+                key={key}
+              >
+                {(message) => <option value={ITEM_STATUS[key]}>{message}</option>}
+              </FormattedMessage>
+            ))}
+          </Select>
+        </div>
+      ),
     };
 
     return (
       <Modal
-        label={<FormattedMessage id="ui-orders.receiving.modalPaneTitle" />}
+        contentClass={css.paneTitle}
+        label={<FormattedMessage id="ui-orders.receiving.modalPaneTitle" values={{ poLineNumber, title }} />}
+        footer={this.getFooter(close)}
         open
       >
-        <Row end="xs">
-          <Button
-            buttonStyle="primary"
-            onClick={close}
-          >
-            <FormattedMessage id="ui-orders.receiving.cancelBtn" />
-          </Button>
-          <Button
-            buttonStyle="primary"
-            disabled
-          >
-            <FormattedMessage id="ui-orders.receiving.receiveBtn" />
-          </Button>
-        </Row>
         <MultiColumnList
           contentData={itemList}
           formatter={resultsFormatter}
-          visibleColumns={['title', 'barcode', 'location', 'itemStatus']}
+          visibleColumns={['isChecked', 'barcode', 'format', 'location', 'itemStatus']}
           columnMapping={{
-            title: <FormattedMessage id="ui-orders.receiving.title" />,
+            isChecked: <Checkbox type="checkbox" checked={this.state.allChecked} onChange={() => this.toggleAll(itemList)} />,
```

Comment:
`itemList` comes from props, no need to pass it into `toggleAll` function, as well as no need to bind callback with arrow function here. It could be `onChange={this.toggleAll}`